test/lint/newline-after-block.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/sh -eu
+
+echo "Checking that functional blocks are followed by newlines..."
+
+files=$(find ./ -type f -name "*.go")
+
+exit_code=0
+for file in $files
+do
+  # This oneliner has a few steps:
+  # 1. sed:
+  #     a. Check for lines that contain a single closing brace (plus whitespace).
+  #     b. Move the pattern space window forward to the next line.
+  #     c. Match lines that start with a word character. This allows for a closing brace on subsequent lines.
+  #     d. Print the line number.
+  # 2. xargs: Print the filename next to the line number of the matches (piped).
+  # 3. If there were no matches, the file name without the line number is printed, use grep to filter it out.
+  # 4. Replace the space with a colon to make a clickable link.
+  RESULT=$(sed -n -e '/^\s*}\s*$/{n;/^\s*\w/{;=}}' "$file" | xargs -L 1 echo "$file" | grep -v '\.go$' | sed 's/ /:/g')
```

Comment:
Is the first address selector allowing a trailing white space after the closing `}`?

Thanks for the comment it helps but I still can't decypher it fully. I ran it on `main` and it didn't catch anything despite having manually removed a `\n` after a function block.

Reply:
@simondeziel yes the first selector is finding any line which contains *only* a single `}` and whitespace. Then, moving the pattern space forward to the next line, we have a match if it begins with a word character.

This should catch any statements that are run on lines immediately following a closing brace:
```golang
if err != nil {
    return err
}
x = 10
```